internal/glfw/gen.go

```diff
@@ -0,0 +1,170 @@
+
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// files that should be compiled
+var filenames = map[string]struct{}{
+	"cocoa_init.m":     {},
+	"cocoa_joystick.m": {},
+	"cocoa_monitor.m":  {},
+	"cocoa_time.c":     {},
+	"cocoa_window.m":   {},
+	"context.c":        {},
+	"egl_context.c":    {},
+	"init.c":           {},
+	"input.c":          {},
+	"monitor.c":        {},
+	"nsgl_context.m":   {},
+	"osmesa_context.c": {},
+	"posix_thread.c":   {},
+	"vulkan.c":         {},
+	"window.c":         {},
+}
+
+type arch string
+
+const (
+	archAmd64 arch = "amd64"
+	archArm64 arch = "arm64"
+)
+
+func (a arch) target() string {
+	switch a {
+	case archAmd64:
+		return "x86_64"
+	case archArm64:
+		return "arm64"
+	default:
+		panic("not reached")
+	}
+}
+
+func execCommand(name string, args ...string) error {
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func objectExt(s string) string {
+	return strings.Trim(strings.Trim(s, ".c"), ".m") + ".o"
+}
+
+func run() error {
+	list := exec.Command("go", "list", "-f", "{{.Dir}}", "github.com/go-gl/glfw/v3.3/glfw")
+	output, err := list.Output()
+	if err != nil {
+		return err
+	}
+	source := strings.TrimSpace(string(output))
+	build, err := os.MkdirTemp("", "ebitengine-glfw-*")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		log.Printf("Deleting build files: %s", build)
+		os.RemoveAll(build) // clean up
+	}()
+	csource := source + "/glfw/src"
+	includes := source + "/glfw/include"
+	dir, err := os.ReadDir(csource)
+	if err != nil {
+		return err
+	}
+
+	for _, a := range []arch{archAmd64, archArm64} {
+		log.Printf("Compiling Files for %s\n", a.target())
+		for _, entry := range dir {
+			if _, ok := filenames[entry.Name()]; !ok {
+				continue
+			}
+			name := entry.Name()
+			//arch := archAmd64
+			args := []string{
+				"-o", // output
+				objectExt(filepath.Join(build, name)),
+				"-arch",
+				a.target(),
+				"-c", // compile without linking
+				filepath.Join("-I", includes, "include/GLFW/glfw3.h"),       // include glfw3.h
+				filepath.Join("-I", includes, "include/GLFW/glfw3native.h"), // include glfw3native.h
+				"-D_GLFW_COCOA",              // define _GLFW_COCOA
+				"-D_GLFW_BUILD_DLL",          // define _GLFW_BUILD_DLL to build dylib
+				"-Os",                        // optimize for size
+				"-g",                         // debug info
+				filepath.Join(csource, name), // .c or .m file
+			}
+			err = execCommand("clang", args...)
+			if err != nil {
+				return err
+			}
+		}
+		doto, err := filepath.Glob(filepath.Join(build, "*.o"))
+		if err != nil {
+			return err
+		}
+		log.Printf("Linking for %s\n", a.target())
+		args := []string{
+			"-dynamiclib",
+			"-Wl",
+			"-arch",
+			a.target(),
+			"-framework", "Cocoa", "-framework", "IOKit", "-framework", "CoreFoundation",
+			"-o",
+			"glfw-" + a.target() + ".dylib",
+		}
```

Comment:
I looked at the g++ documentation `g++ -help` and there is no `,--no-insert-timestamp` option.

Reply:
https://sourceware.org/binutils/docs-2.30/ld/Options.html

Apparently this is an option for (i386) PE, so never mind.